### ISSUE TITLE
Forward attachments both ways

### ISSUE THREAD

**Issue by skimpel** [axel-angel/whatsapp-email-bridge#4]
Do you plan to enhance the script to be able to forward attachments ?

**Comment by axel-angel** [comment]
Yes, well I just hadn't had the request until now, so I can give it a try.

For receiving: it shouldn't be too complicated because WhatsApp send a link to the file. I could just append this link to the email body, this would be the simplest solution. To send: It should be easy to get an email attachement, the question is how hard is it with yowsup to upload file to WhatsApp and send it, I'll take a look.

Would such solution be acceptable?

**Comment by skimpel** [comment]
Any solution would be welcome ! Sending the attachment would be better i think (i'm using outlook as email client ) but a link is better than nothing !

**Comment by axel-angel** [comment]
Media messages are forwarded by email. Click the URL to view it. That's a very basic solution, the next step is to download the media and attach it to the email. Then forwarding medias from email to WhatsApp.

**Comment by skimpel** [comment]
I tried the new version an forwarding to email works with the url links. Forwarding from email to whatsApp gave me an error whatever i tried ... ! Maybe because i have to change the script from LMTPServer to SMTPServer because i'm unable to use local sockets ... (i'm receving media type application/octet. Temp file is created find and readable ! ) ?

**Comment by axel-angel** [comment]
Yes, there is a problem with yowsup, there are lots of bug at the moment, there are some ugly workarounds because yowsup crashes (file type detection, etc). A bit a work is required and probably an upstream pull request will be necessary.

**Comment by skimpel** [comment]
I tried your last modifications. Keep in mind that i had to change connection from LMTP to SMTP to get mails bridged ... ! 
My current state : 
I am able to forward only-text messages created in Outlook. Only problem is converting windows format to whats app ( e.g. getting =0D=0A in WhatsApp ). When formatting as html the message body is sent as attachment type "text/html". There is a problem in the script with pl.get('Content-Type', ...). Only if i change to m.get_content_type() it works.
Sending an attachment does not work for me. An attached jpeg has attachment type "application/octct-stream". So i had to add another type "application" with MEDIA_TYPE_IMAGE. Then the temp file is saved correctly, but upload to whatsapp fails.

Sending messages and images from whatsapp to mail works as you described.

WhatsApp status is not maintained.

**Comment by skimpel** [comment]
Again i tried your last update.
1.) SMTP forwarding works fine for me.
2.) WhatsApp status is not updated.
3.) Text mails are not correctly forwarded ...

```
Date: Thu, 25 Jun 2015 11:01:47 +0200
Subject: Re: Test
X-Mailer: Kerio Connect 8.5.0/Kerio Connect Client
X-User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101
    Firefox/38.0
Message-ID: <579986718-3020@mail.xxx.xxx>
From: Stefan Kimpel <xxx@xxx.xxx>
To: None <whatsapp+xxx@xxx.xxx>
X-Priority: 3
Importance: Normal
In-Reply-To: <579221031-952@mail.xxx.xxx>
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="=-znkJ8dlrhQ4RraEnTmez"

--=-znkJ8dlrhQ4RraEnTmez
Content-Type: text/plain; charset="utf-8"

TTTRRRRR

--=-znkJ8dlrhQ4RraEnTmez
Content-Type: text/html; charset="utf-8"

<html><head></head><body>TTTRRRRR<br></body></html>
--=-znkJ8dlrhQ4RraEnTmez--
```

WhatsApp :

TTTRRRRR=0D=0A
1. ) Media Attachments are not forwarded

<= Mail: Skip unsupported attachment type application/octet-stream; name=image.jpg

```
Date: Thu, 25 Jun 2015 11:07:15 +0200
Subject: Re: Media: image
X-Mailer: Kerio Connect 8.5.0/Kerio Connect Client
X-User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101
    Firefox/38.0
Message-ID: <580314375-1544@mail.xxx.xxx>
From: Stefan Kimpel <s.kimpel@xxx.xxx>
To: None <whatsapp+xxx@xxx.xxx>
X-Priority: 3
Importance: Normal
In-Reply-To: <580245062-4092@mail.xxx.xxx>
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="=-GMSEznwxxULr8jddoP79"

--=-GMSEznwxxULr8jddoP79
Content-Type: multipart/alternative; boundary="=-wbFRoD6DB406m/+oc3Wq"

--=-wbFRoD6DB406m/+oc3Wq
Content-Type: text/plain; charset="utf-8"



--=-wbFRoD6DB406m/+oc3Wq
Content-Type: text/html; charset="utf-8"

<html><head></head><body><br></body></html>
--=-wbFRoD6DB406m/+oc3Wq--

--=-GMSEznwxxULr8jddoP79
Content-Type: image/jpeg; name="Bild.jpg"
Content-Disposition: attachment; filename="Bild.jpg"
Content-Transfer-Encoding: base64

/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcU
FhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgo
KCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAMzAnYDASIA
```

Is it a configuration problem on my mail server ?

Thanks in advance !

**Comment by axel-angel** [comment]
Thanks for the examples, I'll reproduce your problem and it will be easier to fix. It's probably not a mail server problem.

For (2), please continue in issue #5 as I don't know how to address the problem.
For (3), please continue in issue #6 where the problem is addressed.
For (4), it will be addressed here, it's high priority.

PS: I've removed your mail address and phone number for privacy.
